truffleruby/src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -1085,4 +1088,32 @@ protected ObjectIVarSetNode createObjectIVarSetNode() {
 
     }
 
+    @CoreMethod(names = "store_exception", onSingleton = true, needsBlock = true)
+    public abstract static class StoreExceptionNode extends YieldingCoreMethodNode {
+
+        @Specialization
+        public Object executeWithProtect(VirtualFrame frame, DynamicObject block) {
+            try {
+                yield(frame, block);
+                return nil();
+            } catch (Throwable e) {
+                return e;
```

Comment:
This should have 2 BranchProfile to correctly profile whether an exception is thrown or not, even when everything cannot be inlined to prove whether an exception can happen.